csp_billing_adapter/bill_utils.py

```diff
@@ -48,17 +53,25 @@ def get_max_usage(metric: str, usage_records: list) -> int:
     if not usage_records:
         return 0
 
-    max_usage = max(record.get(metric, 0) for record in usage_records)
+    max_usage = max(
+        record.get(metric, 0) for record in usage_records
+        if record['reporting_time'] <= bill_time
```

Comment:
Is there a safety fallback somewhere if the reset of the records fails?

Reply:
Fallbacks, error handling and retries are last on the docket. There can be a secondary check every time the adapter runs to clean up old entries based on last bill time. Last bill time can also be included in calculation.

Reply:
This will be covered by boudning the records on start and end date.